build_runner_core/lib/src/package_graph/apply_builders.dart

```diff
@@ -366,10 +366,13 @@ bool _shouldApply(
   if (builderConfig?.isEnabled != null) {
     return builderConfig.isEnabled;
   }
-  return builderApplication.filter(node.package) ||
-      (applyWith[builderApplication.builderKey] ?? const []).any(
-          (anchorBuilder) =>
-              _shouldApply(anchorBuilder, node, applyWith, allBuilders));
+  if (node.target.autoApplyBuilders) {
```

Comment:
I actually wonder if the logic should be a bit different here.

If a builder is explicitly enabled and applies other builders, should those also be enabled?

Reply:
cc @evanweible-wf what would you expect here?

For instance a source_gen `SharedPartBuilder` needs to apply the `source_gen:combining_builder`.

However enabling that automatically could cause issues if you have multiple source_gen builders you want to enable  on different targets... (each target would try to enable the same builder and then you would get duplicate outputs and have to explicitly disable it again...)

Reply:
It would be nice to only have to to enable/configure the known builders and allow them to apply their own transitive builders.. just to make sure I'm understanding this, the problem you mentioned would only arise if we were splitting source gen builders across targets like so:

```
    module_bootstrap_json:
      builders:
        # disable all builders except json_serializable
    module_bootstrap_built:
      builders:
        # disable all builders except built_value_generator

    # etc..
```

Is that right?

Reply:
Correct, for the source_gen example it would only matter if splitting builders across targets.

It might be reasonable though to say that you just shouldn't do that as it explicitly means you are duplicating some sources across targets which we don't really want people to do (but we don't enforce).

Reply:
+1 We should still enable the builders that get applied for reasons other than their `auto_apply:` config.

Reply:
Seems like we are on the same page here I will update the logic.

I think making this only affect builders that are applied via `auto_apply` is a good way to think about it.